flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBNativeMetricMonitor.java

```diff
@@ -231,4 +252,88 @@ public void update() {
             setStatistics(this);
         }
     }
+
+    /**
+     * A gauge which periodically pulls a RocksDB statistics-based native statistic metric for the database.
+     */
+    class RocksDBNativeHistogramStatisticsMetricView extends RocksDBNativeView implements Histogram {
+        private final HistogramType histogramType;
+
+        private HistogramData histogramData;
+
+        private RocksDBNativeHistogramStatisticsMetricView(HistogramType histogramType) {
+            this.histogramType = histogramType;
+        }
+
+        void setValue(HistogramData histogramData) {
+            this.histogramData = histogramData;
+        }
+
+        @Override
+        public void update() {
+            setHistogramStatistics(this);
+        }
+
+        @Override
+        public void update(long value) {
+        }
+
+        @Override
+        public long getCount() {
+            return histogramData == null ? 0 : histogramData.getCount();
+        }
+
+        @Override
+        public HistogramStatistics getStatistics() {
+            return new RocksDBNativeHistogram();
```

Comment:
The returned statistics should be a snapshot, however, current returned `HistogramStatistics` would change during the time as the underlying histogramData could also change.